nativeunwind/elfunwindinfo/elfgopclntab.go

```diff
@@ -260,6 +260,11 @@ func searchGoPclntab(ef *pfelf.File) ([]byte, error) {
 			continue
 		}
 
+		// Skip segments that are too small anyway.
+		if p.Memsz < uint64(PclntabHeaderSize()) {
```

Comment:
`Memsz` is the size allocated from memory. `Filesz` is the size on disk with data contents. `Memsz` is larger than `Filesz` based on how much zero initialized memory should be allocated in the end. Since `.gopclntab` header should follow with initialized data, `Filesz` would be a more appropriate check.

```suggestion
		if p.Filesz < uint64(PclntabHeaderSize()) {
```